analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldWrittenFrom.cs

```diff
@@ -97,19 +99,24 @@ public abstract class StaticFieldWrittenFrom : SonarDiagnosticAnalyzer
                             var message = GetDiagnosticMessageArgument(cbc.CodeBlock, cbc.OwningSymbol, fieldWithLocations.Key);
                             var secondaryLocations = fieldWithLocations.Key.DeclaringSyntaxReferences
                                                                        .Select(x => x.GetSyntax().GetLocation());
-                            c.ReportIssue(Diagnostic.Create(SupportedDiagnostics[0], location,
+                            c.ReportIssue(Diagnostic.Create(Rule, location,
                                 additionalLocations: secondaryLocations,
                                 messageArgs: message));
                         }
                     });
                 });
 
-        private static void CollectLocationOfStaticField(ExpressionSyntax expression, MultiValueDictionary<IFieldSymbol, Location> locationsForFields, SyntaxNodeAnalysisContext context)
+        private static void CollectLocationOfStaticField(SemanticModel semanticModel, MultiValueDictionary<IFieldSymbol, Location> locationsForFields, ExpressionSyntax expression)
         {
-            if (context.SemanticModel.GetSymbolInfo(expression) is { Symbol: IFieldSymbol { IsStatic: true } fieldSymbol })
+            if (StaticFieldSymbol(semanticModel, expression) is { } fieldSymbol)
             {
                 locationsForFields.Add(fieldSymbol, expression.GetLocation());
             }
         }
+
+        private static IFieldSymbol StaticFieldSymbol(SemanticModel semanticModel, SyntaxNode node) =>
```

Comment:
could you please add the `Get` prefix here, to underline it's getting it from the SemanticModel for the node?

see https://sonarsource.github.io/rspec/#/rspec/S2365 - when I see "StaticFieldSymbol" I generally think that it should be a cheap call (it's a property-like name), however this invokes the semantic model so it's not cheap